### ISSUE TITLE
Allow GradientDescentOptimizer function to take a Tensor as input as well as float

### ISSUE THREAD

**Issue by harshithapv** [SciSharp/TensorFlow.NET#422]
According to Tensorflow docs, the input to  [GradientDescentOptimizer](https://www.tensorflow.org/versions/r1.14/api_docs/python/tf/train/GradientDescentOptimizer) can be a tensor or floating point.
> learning_rate: A Tensor or a floating point value. The learning rate to use.

But currently, I can only send a float value to the function. It would be great to have the functionality to send a tensor as the input too.

**Comment by harshithapv** [comment]
There is a bug in the current implementation. You need to create a separate _prepare function or have a check in your _prepare() implementation to not convert to tensor when I already pass the tensor.

**Comment by Oceania2018** [comment]
op.convert_to_tensor will return itself if passing a tensor.

**Comment by harshithapv** [comment]
Yes  agree. But, what is being passed is a tensor "lr" from float "_lr" which we never set when we pass a tensor which populates "_lr_t".
public override void _prepare()
        {
            var lr = _call_if_callable(_lr);
            _lr_t = ops.convert_to_tensor(lr, name: "learning_rate");
        }

**Comment by Oceania2018** [comment]
OK, understood, can you PR?
